tests/functional/Dockerfile

```diff
@@ -12,26 +12,22 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-FROM ubuntu:xenial
+FROM ubuntu:bionic
 
 # apt stuff
 RUN apt-get update
-RUN apt-get -y install apt-transport-https ca-certificates curl software-properties-common python-pip nodejs npm python-software-properties supervisor memcached redis-server
+RUN apt-get -y install apt-transport-https ca-certificates curl \
+    software-properties-common python2.7 python-pip \
+    nodejs npm ruby supervisor memcached redis-server
 
 RUN curl -fsSL https://download.docker.com/linux/ubuntu/gpg | apt-key add -
 RUN add-apt-repository "deb [arch=amd64] https://download.docker.com/linux/ubuntu $(lsb_release -cs) stable"
 RUN apt-get update
 RUN apt-get -y install docker-ce
 
-RUN apt-add-repository ppa:brightbox/ruby-ng
-RUN apt-get update
-RUN apt-get -y install ruby2.4
-
 # aws fakes
 RUN gem install fake_sqs -v 0.4.3
-# kinesalite@1.11.5 and above have a duplicate message bug. do not use.
```

Comment:
keep this comment

Reply:
It looked like an obsolete comment... we are not using an old version here. 1.11.5 was released 2 years ago.
Seems awkward to have warnings about things that are no longer applicable.

Reply:
it says kinesalite@1.11.5 **and above** have a duplicate message bug. do not use.

Reply:
Honestly clear on what you're suggesting with the "and above" part here... and versions before that? Versions after? Or are you referring to something above this line in the Dockerfile?

Also not clear on how it applies to what we have set in the Dockerfile right now.

Reply:
'and above' means 'and greater'

Reply:
Oh... so this warning is saying "don't use Kinesalite 1.11.5 or greater"?
The Dockerfile is (and was before this PR) specifying version 1.14.0, which is greater than 1.11.5.